src/org/zaproxy/zap/extension/spider/SpiderThread.java

```diff
@@ -484,7 +509,16 @@ public void run() {
 			return;
 		}
 
-		SessionStructure.addPath(Model.getSingleton().getSession(), historyReference, message);
+        StructuralNode node = SessionStructure.addPath(
+                Model.getSingleton().getSession(), historyReference, message, true);
+		if (node != null) {
+		    try {
+                addUriToAddedNodesModel(SessionStructure.getNodeName(message), 
+                        message.getRequestHeader().getMethod(), "");
+            } catch (URIException e) {
+                log.error("Error while recording added node: " + e.getMessage(), e);
```

Comment:
Without searching the code base for this message text is "recording" going to be meaningful to a user or dev?

Perhaps this should be about adding the node to the tree?